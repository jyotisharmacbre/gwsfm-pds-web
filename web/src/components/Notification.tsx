import React from 'react';
import NotificationsIconImage from '@material-ui/icons/Notifications';
import { IconButton, Badge } from '@material-ui/core';

export default function Notification(props: {NotificationCount:number, handleClick: any}){
   
        return (
           <IconButton color="inherit" onClick={props.handleClick}>
            <Badge badgeContent={props.NotificationCount}  > 
              <NotificationsIconImage id="BellIcon" name="BellIcon" color="secondary"  />
            </Badge>
          </IconButton>
         
          );
   
   
  }
  