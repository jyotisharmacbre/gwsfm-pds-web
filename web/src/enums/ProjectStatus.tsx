enum ProjectStatus {
  InitialCustomerInquiry = 1,
  JA = 2,
  InReview = 3,
  BidLost = 6,
  OnHold = 7,
  OrderReceived = 5,
  JAApproved = 4,
  Live = 9,
  Completed = 10
}

export default ProjectStatus;
