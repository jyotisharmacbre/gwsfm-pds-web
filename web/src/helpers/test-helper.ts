import checkPropTypes from 'check-prop-types';

export const findByTestAtrr = (component: any, attr: string) => {
  const wrapper = component.find(`[data-test='${attr}']`);
  return wrapper;
};

export const checkProps = (component: any, expectedProps: any) => {
  const propsErr = checkPropTypes(
    component.propTypes,
    expectedProps,
    'props',
    component.name
  );
  return propsErr;
};
