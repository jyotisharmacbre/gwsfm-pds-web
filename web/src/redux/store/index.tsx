import { applyMiddleware, createStore } from 'redux';
import { persistReducer, persistStore } from 'redux-persist';
import storage from 'redux-persist/lib/storage';
import thunk from 'redux-thunk';

import allReducer from '../reducers';

const persistConfig = {
  key: 'root',
  storage
};

const persistedReducer = persistReducer(persistConfig, allReducer);

const middleware = applyMiddleware(thunk);
const store = createStore(persistedReducer, middleware);
const persistor = persistStore(store);
export { store, persistor };
