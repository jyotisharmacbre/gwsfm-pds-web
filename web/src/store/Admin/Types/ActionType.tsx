export enum ActionType {
	GET_PROJECT_PARAMETERS_SUCCESS = 'GET_PROJECT_PARAMETERS_SUCCESS',
	GET_PROJECT_PARAMETERS_ERROR = 'GET_PROJECT_PARAMETERS_ERROR'
}
