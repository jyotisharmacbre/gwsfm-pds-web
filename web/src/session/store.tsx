import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger'
import rootReducer from './reducers/RootReducer';

const middlewares: any[] = [];

middlewares.push(thunk);

if (process.env.NODE_ENV === `development`) {

  middlewares.push(logger);
}

export default function configureStore(initialState = {}) {
  return createStore(
    rootReducer,
    applyMiddleware(...middlewares)
  );
}