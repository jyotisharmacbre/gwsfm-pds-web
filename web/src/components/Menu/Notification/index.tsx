import React, { useEffect } from 'react';
import { INotification } from "../../../store/Notifications/Types/INotification";
import IReactIntl from "../../../Translations/IReactIntl";
import { injectIntl } from "react-intl";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faBell } from "@fortawesome/free-solid-svg-icons";
import { useHistory } from "react-router-dom";
import { ILookup } from '../../../store/Lookups/Types/ILookup';


interface IMapStateToProps {
	notifications: Array<INotification>;
	showNotification: boolean;
	lookups: Array<ILookup>;
}


const Notification: React.FC<IMapStateToProps & IReactIntl> = (props) => {
	let history = useHistory();

	//add & remove class for pipeline and dashboard page
	const showClass = () => {
		return history.location.pathname == "/" ||
			history.location.pathname == "/Pipeline"
	}

	const getTemplate = (notificationType) => {
		return props.lookups?.find(x => x.lookupKey == notificationType && x.lookupItem == 'Notification_Template')?.description;
	}

	const getUnreadNotificationCount = () => {
		return props.notifications?.filter(x => x.status === 0).length;
	}

	return (
		<div>
			<i>
				<FontAwesomeIcon className="" icon={faBell} />
				{getUnreadNotificationCount() > 0 && <span className="badge badge-light"></span>}
			</i>
			{getUnreadNotificationCount()>0 ? <div className={`dropdown-menu notify_dropdown user-dropdown ${props.showNotification ? 'show' : 'hide'}
	  ${showClass() ? 'default' : 'all_pages'} `}>
				<ul>
					<li>
						<div className="notify_topbar">
							<span>({getUnreadNotificationCount()}) New Notifications</span>
							<span>Mark all as Read</span>
						</div>
					</li>
					{props.notifications.map(notification => {
						return (<li className="bg-grey">
							<a href={`/ReviewApprove/${notification.projectId}`}>
							<h4 className="title">
								{`${getTemplate(notification.notificationType)}`}
							</h4>
							<p className="brief">
								{notification.description}
							</p>
							</a>
							<span className="mark_sign">MARK READ</span>
						</li>)

					})}
				</ul>
			</div>
			:
     <div className={`dropdown-menu notify_dropdown user-dropdown ${props.showNotification ? 'show' : 'hide'}
	  ${showClass() ? 'default' : 'all_pages'} `}>
				<ul>
					<li>
						<p>No Notification for you</p>
					</li>
				</ul>
			</div>}
		</div>

	);
};


export default injectIntl(Notification);
