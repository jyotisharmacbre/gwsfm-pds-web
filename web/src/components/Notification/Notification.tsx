import React, { useEffect } from 'react';
import NotificationsIconImage from '@material-ui/icons/Notifications';
import { IconButton, Badge } from '@material-ui/core';
import { ThunkDispatch } from 'redux-thunk';
import { AnyAction } from 'redux';
import { connect } from 'react-redux';
import { IApplicationState } from '../../session/rootReducer';
import {  INotificationProps } from '../../props/AppProps';
import { getNotificationActionCreator } from '../../session/Notification/Action';

const mapDispatchToProps = (dispatch: ThunkDispatch<any, any, AnyAction>) => {
  return {
    handleClick: () => { window.location.href = "/Notifications"; },
    getNotificationCount: () => dispatch(getNotificationActionCreator())
  }
};

const mapStateToProps = (state: IApplicationState) => {
  return {
    data: state.notificationState
  }
};

const Notification: React.FC<INotificationProps> = (props) => {

  useEffect(() => {
    props.getNotificationCount();
  }, []);

  return (
    <IconButton color="inherit" onClick={props.handleClick}>
      <Badge badgeContent="5" >
        <NotificationsIconImage id="BellIcon" name="BellIcon" color="secondary" />
        
      </Badge>
    </IconButton>
  );


}

export default connect(mapStateToProps, mapDispatchToProps)(Notification);
