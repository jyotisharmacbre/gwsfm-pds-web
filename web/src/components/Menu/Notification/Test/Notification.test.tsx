import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { findByTestAtrr } from '../../../../helpers/test-helper';
import { IntlProvider } from 'react-intl';
import translations from '../../../../Translations/translation';
import configureStore from 'redux-mock-store';
import { BrowserRouter } from 'react-router-dom';
import ProjectStatus from '../../../../enums/ProjectStatus';
import Notification from '../../Notification/index'
import { store } from '../../../../store';
import { ActionType } from '../../../../store/Notifications/Types/ActionType';
import notificationReducer, { initialState } from '../../../../store/Notifications/Reducer';

let props: any = {
	notifications: [{
		notificationId: 1,
		projectId: 1,
		senderId: "test@pds.com",
		receiverId: "test@pds1.com",
		notificationType: 1,
		description: 1,
		status: 0,
		createdOn: null,
		modifiedOn: null
	}],
	showNotify: true,
	lookUpData: [{ lookupItem: 'Project_Status', lookupKey: 1 }, { description: "test", lookupItem: 'Notification_Template', lookupKey: 1 }, { lookupItem: 'Engagement_Type', lookupKey: 0 }, { lookupItem: 'Asset_Type', lookupKey: 0 }]
};
describe('Notification Renders', () => {
	const mockStore = configureStore([]);
	let store;
	let wrapper;
	const setUpStore = () => {
		store = mockStore({
		});
	};
	const renderComponent = (store, props) => {
		wrapper = mount(
			<Provider store={store}>
				<IntlProvider locale="en" messages={translations['en'].messages}>
					<BrowserRouter>
						<Notification
							notifications={props.notifications}
							showNotification={props.showNotify}
							lookups={props.lookUpData}
						/>
					</BrowserRouter>
				</IntlProvider>
			</Provider>
		);
	};
	beforeEach(() => {
		setUpStore();
	});
	it('Defines the component', () => {
		renderComponent(store, props);
		expect(wrapper).toBeDefined();
	});
	it('should match the snapshot', () => {
		renderComponent(store, props);
		expect(wrapper).toMatchSnapshot();
	});
	it('should have notify_dropdown', () => {
		renderComponent(store, props);
		expect(wrapper.find('.notify_dropdown')).toHaveLength(1);
	});

	it('should have notifications header', () => {
		renderComponent(store, props);
		expect(wrapper.find('.notify_topbar')).toHaveLength(1);
	});

	it('should render list of notifications', () => {
		renderComponent(store, props);
		expect(wrapper.find('.brief')).toHaveLength(1);
	});

	
	it('should have badge when unread notification', () => {			
		renderComponent(store, props);
		console.log(wrapper.debug());
		expect(wrapper.find('.badge')).toHaveLength(1);
	});
	
	it('should not render list of notifications', () => {	
	props.notifications =	[];	
		renderComponent(store, props);
		expect(wrapper.find('.brief')).toHaveLength(0);
	});

	it('should not have bedge when no unread message', () => {	
		props.notifications = [{
			notificationId: 1,
			projectId: 1,
			senderId: "test@pds.com",
			receiverId: "test@pds1.com",
			notificationType: 1,
			description: 1,
			status: 1,
			createdOn: null,
			modifiedOn: null
		}]	
			renderComponent(store, props);
			expect(wrapper.find('.badge')).toHaveLength(0);
		});

		it('should not have bedge when no message', () => {	
			props.notifications =	[];	
				renderComponent(store, props);
				expect(wrapper.find('.badge')).toHaveLength(0);
			});

});

describe('Notification reducer', () => {
	beforeEach(() => {
	  store.dispatch = jest.fn();
	})
  
	it('should handle get notification success action', () => {
	  const getAllNotifications: any = {
		type: ActionType.GET_ALL_NOTIFICATIONS_SUCCESS,
		payload: [{}],
	  };
	  expect(
		notificationReducer(initialState, getAllNotifications)
	  ).toMatchSnapshot();
	});
  
	it('should handle get notifications failure', () => {
	  const getNotificationFailureAction: any = {
		type: ActionType.GET_ALL_NOTIFICATIONS_ERROR,
		error: true
	  };
	  expect(
		notificationReducer(initialState, getNotificationFailureAction)
	  ).toMatchSnapshot();
	});
  
});
