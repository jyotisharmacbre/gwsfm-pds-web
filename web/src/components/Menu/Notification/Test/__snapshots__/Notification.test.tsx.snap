// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification Renders should match the snapshot 1`] = `ReactWrapper {}`;

exports[`Notification reducer should handle get notification success action 1`] = `
Object {
  "error": null,
  "loading": false,
  "notifications": Array [
    Object {},
  ],
}
`;

exports[`Notification reducer should handle get notifications failure 1`] = `
Object {
  "error": undefined,
  "loading": false,
  "notifications": Array [],
}
`;
